x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/read/ValuesFromManyReader.java

```diff
@@ -85,7 +85,7 @@ void run(int offset) throws IOException {
                  * loading in a way that is correct for the mapped field type, and then convert between that type and the desired type.
                  */
                 fieldTypeBuilders[f] = operator.fields[f].info.type().newBlockBuilder(docs.getPositionCount(), operator.blockFactory);
-                buildersAndLoaders.add(new HashMap<>());
+                buildersAndLoaders.add(new LinkedHashMap<>()); // use LinkedHashMap to preserve insertion order
```

Comment:
Ideally, we should replace the Map with a List, as we read shards in monotonically increasing order. I will follow up to remove this map soon.